src/nbl/video/IOpenGL_LogicalDevice.h

```diff
@@ -625,6 +666,27 @@ class IOpenGL_LogicalDevice : public ILogicalDevice, protected impl::IOpenGL_Log
                 incrementProgressionSync(_gl);
             }
                 break;
+            case ERT_ALLOCATE:
+            {
+                auto& p = std::get<SRequestAllocate>(req.params_variant);
+                IDeviceMemoryAllocator::SMemoryOffset* pretval = reinterpret_cast<IDeviceMemoryAllocator::SMemoryOffset*>(req.pretval);
+                IDeviceMemoryAllocator::SMemoryOffset& retval = *pretval;
+                if(p.allocateInfo.dedication)
+                {
+                    // TODO(Erfan): Use COpenGLMemoryAllocation to include COpenGLImage too and static cast to that and use the virtual funtion initMemory
+                    COpenGLBuffer* glBuffer = static_cast<COpenGLBuffer*>(p.allocateInfo.dedication);
+                    glBuffer->initMemory(&gl, p.memoryTypeIndexFlags);
+                    retval.memory = core::smart_refctd_ptr<COpenGLBuffer>(glBuffer);
+                    retval.offset = 0ull;
```

Comment:
you can set the memory type bits, etc in the `allocate` directly

Just make sure `initMemory` does no OpenGL calls, and you don't need to do them on a thread